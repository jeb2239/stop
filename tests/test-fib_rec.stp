/*
 * test-fib_rec.stp
 * ====================
 */

def fib = (i:Int):Int {
	printf("%d\n", i);
	return 0;
	/*
	if (i > 0) {
		return 1;
	} else {
		return 0;
	}
	} else if (i == 1) {
		return 1;
	} else {
		return fib(i-1) + fib (i-2);
	}
	*/
}

def main = ():Int {
	/*
	printf("%d\n", fib(10));
	printf("%d\n", fib(9));
	printf("%d\n", fib(8));
	*/
	printf("%d\n", fib(1));
	printf("%d\n", fib(0));
	return 0;
}
