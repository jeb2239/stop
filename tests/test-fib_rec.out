55
34
21
